http-server/src/Main.gren

```diff
@@ -0,0 +1,75 @@
+module Main exposing (main)
+
+import Node
+import Bytes exposing (Bytes)
+import Bytes.Encode as BE
+import Node.Program as Program exposing (Program)
+import Node.Http as Http
+
+
+main : Program Model Msg
+main =
+    Program.define
+        { init = init
+        , update = update
+        , subscriptions = subscriptions
+        }
+
+
+type alias Model =
+    { serverKey : Http.ServerKey
+    }
+
+
+type Msg
+    = HandleRequest Http.Request
+    {-
+        type alias Request =
+            { headers : Dict String String
+            , body : Bytes
+            , url : Url
+            , method : Method
+            }
+        
+        type Method
+            = GET
+            | HEAD
+            | POST
+            | PUT
+            | DELETE
+            | CONNECT
+            | TRACE
+            | PATCH
+    -}
+
+
+init : Program.AppInitTask { model : Model, command : Cmd Msg }
+init =
+    Program.await Http.createServer <| \serverKey ->
+        Program.startProgram
+            { model =
+                { serverKey = serverKey
+                }
+            , command =
+                Cmd.none
+            }
+
+
+update : Msg -> Model -> { model : Model, command : Cmd Msg }
+update msg model =
+    case msg of
+        HandleRequest req ->
+            { model = model
+            , command = Http.respondWith model.serverKey
+                { status = 200
+                , body = BE.encode (BE.string "<html>Hello</html>")
+                , headers = Dict.fromArray
+                    [ { key = "Content-type", value = "text/html" }
+                    ]
+                }
+            }
+
+
+subscriptions : Model -> Sub Msg
+subscriptions model =
+    Http.onRequest model.serverKey HandleRequest
```

Comment:
In the future, we might have different subscription functions that specialises over the requests `body` type. Examples `Http.onJsonRequest` or `Http.onStreamRequest`.

Reply:
would it be better for that to be handled with a custom type on body?

Reply:
I guess it depends. But for `Stream` and `Byte` specifically: `Stream` represents an asynchronous stream of bytes. You cannot just call a function on it and get JSON out, you'll need to handle it in a task or through some other means. For `Byte` however, you can convert it directly to JSON.